lib/dedupe/map.go

```diff
@@ -75,5 +86,37 @@ func (f *ForgetfulSyncMap) AddKey(key interface{}) {
 			return
 		}
 	}
-	f.lookup.Store(key, time.Now())
+	f.lookup.Store(key, ForgetableItem{
+		age: time.Now(),
+	})
+}
+
+func (f *ForgetfulSyncMap) Load(key interface{}) (value interface{}, ok bool) {
+	retVal, retBool := f.lookup.Load(key)
+
+	if retVal != nil {
+		return retVal.(ForgetableItem).value, retBool
+	} else {
+		return retVal, retBool
+	}
+}
+
+func (f *ForgetfulSyncMap) Store(key, value interface{}) {
+	f.lookup.Store(key, ForgetableItem{
+		age:   time.Now(),
+		value: value,
+	})
+}
+
+func (f *ForgetfulSyncMap) Delete(key interface{}) {
+	f.lookup.Delete(key)
+}
+
+func (f *ForgetfulSyncMap) Len() (entries int32) {
+	f.lookup.Range(func(key interface{}, value interface{}) bool {
```

Comment:
Consider using a counter for this instead of an expensive Range() call each time.